tests/test_main.py

```diff
@@ -1,11 +1,125 @@
+import uuid
+import pytest
 from fastapi.testclient import TestClient
-
 from src.main import app
+from typing import Generator
+
+from src.schemas import Project, ProjectDetails
+
+
+@pytest.fixture
+def client() -> Generator[TestClient, None, None]:
+    yield TestClient(app)
+
+
+@pytest.fixture(autouse=True)
+def clear_projects() -> Generator[None, None, None]:
+    from src.main import projects
+
+    projects.clear()
+    yield
+    projects.clear()
+
+
+@pytest.fixture
+def sample_project() -> Project:
+    return Project(name="Project 1", description="Test project")
+
+
+@pytest.fixture
+def sample_project_2() -> Project:
+    return Project(name="Project 2", description="Another test project")
+
+
+@pytest.fixture
+def created_project(sample_project: Project) -> ProjectDetails:
+    from src.main import projects
+
+    project_details = ProjectDetails(**sample_project.model_dump())
+    projects.append(project_details)
+    return project_details
+
+
+@pytest.fixture
+def created_projects(client: TestClient, sample_project: Project, sample_project_2: Project) -> list[ProjectDetails]:
+    projects = []
+    for project in [sample_project, sample_project_2]:
+        response = client.post("/projects", json=project.model_dump())
```

Comment:
do not use api to save data for your tests, I agree on such approach for full integration tests, but for now lets keep it simple as you did in created_project fixture